src/NumericalAlgorithms/Interpolation/CubicSpline.hpp

```diff
@@ -0,0 +1,60 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <gsl/gsl_spline.h>
+#include <vector>
+
+/// \cond
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace intrp {
+/*!
+ * \ingroup NumericalAlgorithmsGroup
+ * \brief A cubic spline interpolation class
+ *
+ * The class builds a cubic spline interpolant using the `x_values` and
+ * `y_values` passed into the constructor. For details on the algorithm see
```

Comment:
do the `x_values` and `y_values` need to be sorted? If so there should be an ASSERT that verifies that.

Reply:
yep good point, I'll add that